### PR TITLE
rijks project part 1

### PR DESCRIPTION
**Pull request by LeoraBennett**

### CODE REVIEW

File: src/main/java/bennett/rijksmuseum/ApiKey.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+import com.andrewoid.EnvironmentHandler;
+import com.andrewoid.PropertiesHandler;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+import java.util.logging.Handler;
+
+public class ApiKey {
```
Comment (andrewoid) [inline src/main/java/bennett/rijksmuseum/ApiKey.java]: You didn't need this anymore. 

Instead it's added as a dependency.

File: src/main/java/bennett/rijksmuseum/RijksMuseumService.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import io.reactivex.rxjava3.core.Single;
+import retrofit2.http.GET;
+import retrofit2.http.Query;
+
+/**
+ * https://www.rijksmuseum.nl/en
+ */
+public interface RijksMuseumService {
+
+    // api key = jb32sYqU
```
Comment (andrewoid) [inline src/main/java/bennett/rijksmuseum/RijksMuseumService.java]: You were told NOT to check the API key into github

File: src/test/java/bennett/rijksmuseum/RijksMuseumServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        @Override
+        public Single<CurrentCollection> getFromQuery(String appId, String artist, String pageNumber) {
+            // Simulate API call and return a mock response
+            CurrentCollection mockCollection = new CurrentCollection();
+            return Single.just(mockCollection);
+        }
+    }
+
+    @Test
+    public void testCurrentCollection() {
```
Comment (andrewoid) [inline src/test/java/bennett/rijksmuseum/RijksMuseumServiceTest.java]: Do not put the word test in the method name

File: src/test/java/bennett/rijksmuseum/RijksMuseumServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        RijksMuseumService service = new TestRijksMuseumService();
+
+        Single<CurrentCollection> single = service.currentCollection("jb32sYqU", "1");
```
Comment (andrewoid) [inline src/test/java/bennett/rijksmuseum/RijksMuseumServiceTest.java]: You need to use ApiKey to retrieve this.

File: src/test/java/bennett/rijksmuseum/RijksMuseumServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+import bennett.rijksmuseum.json.CurrentCollection;
+import io.reactivex.rxjava3.core.Single;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class RijksMuseumServiceTest {
+
+    private static class TestRijksMuseumService implements RijksMuseumService {
```
Comment (andrewoid) [inline src/test/java/bennett/rijksmuseum/RijksMuseumServiceTest.java]: This is not how I told you to test services.

File: src/main/java/bennett/rijksmuseum/json/ArtImage.java [hunk 6]
```diff
@@ -0,0 +1,5 @@
+package bennett.rijksmuseum.json;
+
+public class ArtImage {
+
```
Comment (andrewoid) [inline src/main/java/bennett/rijksmuseum/json/ArtImage.java:4]: This has a url

File: src/test/java/bennett/rijksmuseum/RijksMuseumServiceTest.java (hunk 5 above)
Comment (LeoraBennett) [reply src/test/java/bennett/rijksmuseum/RijksMuseumServiceTest.java]: I can't figure out how to do it properly, I rewatched the whole weather video but still couldnt get there. How should I do it?

File: src/main/java/bennett/rijksmuseum/json/ArtImage.java (hunk 6 above)
Comment (LeoraBennett) [reply src/main/java/bennett/rijksmuseum/json/ArtImage.java:4]: which url?

Comment (andrewoid) [reply src/main/java/bennett/rijksmuseum/json/ArtImage.java:4]: The one in the json

File: src/test/java/bennett/rijksmuseum/RijksMuseumServiceTest.java (hunk 5 above)
Comment (andrewoid) [reply src/test/java/bennett/rijksmuseum/RijksMuseumServiceTest.java]: Use the factory and create a service...

File: src/test/java/bennett/rijksmuseum/RijksMuseumServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+            CurrentCollection mockCollection = new CurrentCollection();
+            return Single.just(mockCollection);
+        }
+    }
+
+    @Test
+    public void currentCollection() {
+        RijksMuseumService service = new TestRijksMuseumService();
+
+        Single<CurrentCollection> single = service.currentCollection(ApiKey.APIKEY, "1");
```
Comment (andrewoid) [inline src/test/java/bennett/rijksmuseum/RijksMuseumServiceTest.java]: You need to pass the key that is given to you by Rijks.

My GitHub page for apikeys shows you how to use the ApiKey class

File: src/test/java/bennett/rijksmuseum/RijksMuseumServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @Test
+    public void getFromPageNumber() {
+        RijksMuseumService service = new TestRijksMuseumService();
+
+        Single<CurrentCollection> single = service.getFromPageNumber(ApiKey.APIKEY, "Rembrandt");
+
+        // Assert that the returned Single is not null
+        assertEquals(true, single != null);
```
Comment (andrewoid) [inline src/test/java/bennett/rijksmuseum/RijksMuseumServiceTest.java]: You need to assert that the CurrentCollection holds actual data

File: src/test/java/bennett/rijksmuseum/RijksMuseumServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @Test
+    public void getFromPageNumber() {
+        RijksMuseumService service = new TestRijksMuseumService();
+
+        Single<CurrentCollection> single = service.getFromPageNumber(ApiKey.APIKEY, "Rembrandt");
```
Comment (andrewoid) [inline src/test/java/bennett/rijksmuseum/RijksMuseumServiceTest.java]: You are supposed to be using blockingGet()

File: build.gradle [hunk 10]
```diff
@@ -10,8 +10,11 @@ repositories {
 }
 
 dependencies {
-    testImplementation platform('org.junit:junit-bom:5.9.1')
-    testImplementation 'org.junit.jupiter:junit-jupiter'
```
Comment (andrewoid) [inline build.gradle]: Why did you remove this?
This is what allows you to run tests.

File: src/main/java/bennett/rijksmuseum/RijksMuseumService.java [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+    Single<CurrentCollection> getFromQuery(
+            @Query("key") String apiKey,
+            @Query("q") String query,
+            @Query("p") int pageNumber
+    );
+
+    @GET("/api/en/collection")
+    Single<CurrentCollection> getFromArtist(
+            @Query("key") String apiKey,
+            @Query("q") String query,
```
Comment (andrewoid) [inline src/main/java/bennett/rijksmuseum/RijksMuseumService.java]: If you are passing the artist, you don't need this param

File: src/test/java/bennett/rijksmuseum/RijksMuseumServiceTest.java [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+                keyString,
+                "",
+                2,
+                "Michael Angelo"
+        ).blockingGet();
+
+
+        // then
+        assertNotNull(currentCollection);
+        assertNotNull(currentCollection.getArtObjects());
```
Comment (andrewoid) [inline src/test/java/bennett/rijksmuseum/RijksMuseumServiceTest.java]: Assert that the first element in the array has data inside it.

**Comment by LeoraBennett** [comment]
My getFromArtist test is failing and I can't figure out how to fix it, what should I do?

**Review by andrewoid — APPROVED** [review]
